### ISSUE TITLE
Kjøreplan validering slår ut på etteranmeldinger

### ISSUE THREAD

**Issue by heming-h** [langrenn-sprint/race-service#138]
Når en løper etteranmeldes er det ikke nødvendigvis feil - enten bør valideringen endres eller så bør vi ha ulike alvorlighetsgrader på velideringesfeil. Kanskje dele opp i 3 kategorier - info, warning og error?
0: Number of contestants in raceplan (668) is not equal to the number of contestantsin the raceclasses (672).

**Comment by stigbd** [comment]
Hmmm. Interessant. Når oppstår denne feilen? Kva må til for å gjenskape?

Eg er ellers litt usikker på korleis etteranmeldings-logikken fungerer. En etteranmeldt løper bør jo leggast inn i raceplan/race.

**Comment by heming-h** [comment]
Ja, vi må snakke om hele raceplan-valideringen. Jeg synes den blir for teknisk orientert og forteller brukeren om saker som ikke er et problem i praksis. De tekniske begrensningene må backend selv passe på. For meg var hensikten å hjelpe brukeren i å unngå feil. Et par eksempel på hva jeg kunne tenke meg i validering: (1) Hjelpe til med å få et optimalt oppsett i forhold til tidsskjema og hviletider. (2) Si noe om hvor mange etteranmeldinger det er plass til og evt. i hvilke heat.

**Comment by stigbd** [comment]
> Ja, vi må snakke om hele raceplan-valideringen. Jeg synes den blir for teknisk orientert og forteller brukeren om saker som ikke er et problem i praksis. De tekniske begrensningene må backend selv passe på.

Litt usikker på kva du legger i "tekniske begrensninger". For meg, som admin av et race, har valideringa vore svært nyttig. Eg har oppdaga feil både i front-end og back-end funksjonalitet.

> For meg var hensikten å hjelpe brukeren i å unngå feil. Et par eksempel på hva jeg kunne tenke meg i validering: (1) Hjelpe til med å få et optimalt oppsett i forhold til tidsskjema og hviletider.

Enig. Denne ligg som issue https://github.com/langrenn-sprint/race-service/issues/118

> (2) Si noe om hvor mange etteranmeldinger det er plass til og evt. i hvilke heat.

Enig i at dette er nyttig. Men usikker på om det er smart å plassere denne informasjonen inn i kjøreplan-viewet. Vi kunne vurdere å lage et eget etteranmelings "dashboard".

> (3) løpere som er plassert i heat med en annen løpsklasse.

Antat at du meiner "en annen løpsklasse enn løperens aldersklasse". Enig i at dette er nyttig. Lager issue.

> (4) Løpere som står plassert i flere heat.

Enig i at dette er nyttig. Lager issue.

Men eg ville ha sagt at (3) og (4) høyrer betre heime i ein "valider startliste" funksjon. Dette er tunge spørringer som gjer raceplan-validering endå meir krevande.

**Comment by heming-h** [comment]
Det blir også validerjingsfeil når starter i semifinale / finale opprettes - eksempel på melding etter at en klasse er ferdig kjørt på renndagen: 

0: The sum of contestants in races (80) is not equal to the number of contestants in the raceplan (112).Number of contestants in raceplan (112) is not equal to the number of contestants in the raceclasses (80).
--


<br class="Apple-interchange-newline" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-size-adjust: auto; -webkit-text-stroke-width: 0px; text-decoration: none;">

**Comment by stigbd** [comment]
Bra funn. Har fiksa dette no. Topp om du kan reteste.

**Comment by heming-h** [comment]
Testet OK - lukker.
